### ISSUE TITLE
EnC: Moving a method with a parameter across partial definitions issues wrong rude edit

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#52211]
**Version Used**: 
Version 16.10.0 Preview 2.0 [31128.4.main]

**Steps to Reproduce**:

Before edit:

File 1:
```C#
partial class C
{
   void F2() {}
}
```

File 2:
```C#
partial class C
{
   void F2(int a) {}
}
```

After edit:

File 1:
```C#
partial class C
{
   void F2(int a) {}
}
```

File 2:
```C#
partial class C
{
   void F2() {}
}
```

**Expected Behavior**:

Each document has semantic update of F2, respectively.

**Actual Behavior**:

"Add parameter" rude edit is reported.

**Comment by davidwengier** [comment]
FYI @tmat, not sure if you want to just consider this another part of #51011
